### PR TITLE
frr: fix cross compilation and add advanced options

### PR DESCRIPTION
**Pull request by thillux**
###### Description of changes

FRR did not cross-compile from x86-64 to aarch64, because the cli interface helper clippy, needs to be compiled with the build host toolchain. After some failed experiments inside default.nix, I concluded, that it would we cleaner to add a small helper package, which can the be built with the native toolchain and reference clippy from there. As this is frr internal, I did not expose it to all-packages.nix.

SNMP is used with net-snmp-config to obtain the compilation flags. It did not get this working without binfmt, as it either is called in the native form and returns e.g. x86-64 libs, which do no link on aarch64 or it is in aarch64 form, which does not execute without binfmt. Therefore I made snmp support conditional on a working host or binfmt toolchain.

For networking purposes the exposed options of the package where to small, I added most networking relevant options, like route replacement semantics and made it possible to disable most of the daemons if hardening is a need (e.g. only BGP supported). 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by thillux** [comment]
Result of `nixpkgs-review pr 244670` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>frr</li>
  </ul>
</details>

File: pkgs/servers/frr/default.nix
Comment (thillux) [reply pkgs/servers/frr/default.nix]: Thanks for the suggestion. I changed accordingly.

File: pkgs/servers/frr/default.nix
Comment (RaitoBezarius) [inline pkgs/servers/frr/default.nix:122]:
```suggestion
  clippy-helper = buildPackages.callPackage ./clippy-helper.nix { frrVersion = version; frrSource = src; };
```

File: pkgs/servers/frr/clippy-helper.nix
Comment (RaitoBezarius) [inline pkgs/servers/frr/clippy-helper.nix:21]:
```suggestion
  src = frrSource;
```

File: pkgs/servers/frr/clippy-helper.nix
Comment (RaitoBezarius) [inline pkgs/servers/frr/clippy-helper.nix:19]:
```suggestion
  version = frrVersion;
```

**Review by RaitoBezarius — COMMENTED** [review]
LGTM modulo some minor changes

**Review by RaitoBezarius — APPROVED** [review]

File: pkgs/servers/frr/default.nix
Comment (thillux) [reply pkgs/servers/frr/default.nix]: I added a comment and moved the check from the assertion to the default parameters in combination with a warning.

**Comment by RaitoBezarius** [comment]
Waiting for CI, that's pretty good already, thanks!

**Comment by thillux** [comment]
@RaitoBezarius

**Pull request merged**
